Invoke-IcingaCheckUsedPartitionSpace - problem with mirrors disks

Hi, 
the cmdlet Invoke-IcingaCheckUsedPartitionSpace throws exceptions when you have mirrored disks.

(even with version 1.5.0)


```
PS C:\Windows\system32> Invoke-IcingaCheckUsedPartitionSpace
Ausnahme beim Aufrufen von "Add" mit 2 Argument(en):  "Das Element wurde bereits hinzugefügt. Schlüssel im Wörterbuch:
"3". Hinzuzufügender Schlüssel: "3"."
In C:\Program
Files\WindowsPowerShell\Modules\icinga-powershell-plugins\provider\disks\Get-IcingaDiskPartitionAssignment.psm1:53
Zeichen:9
+         $PartitionInformation[$DiskId].Partitions.Add(
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException

Ausnahme beim Aufrufen von "Add" mit 2 Argument(en):  "Das Element wurde bereits hinzugefügt. Schlüssel im Wörterbuch:
"3". Hinzuzufügender Schlüssel: "3"."
In C:\Program
Files\WindowsPowerShell\Modules\icinga-powershell-plugins\provider\disks\Get-IcingaDiskPartitionAssignment.psm1:53
Zeichen:9
+         $PartitionInformation[$DiskId].Partitions.Add(
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ArgumentException

[OK] Used Partition Space: 4 Ok
| 'partition_d'=656880300000B;;;0;852476900000 'partition_c'=62066870000B;;;0;107375200000
```

![image](https://user-images.githubusercontent.com/2630551/120603279-227b5a00-c44c-11eb-98b5-ea7b3b509b60.png)

I think it's the same problem like issue #130

the PR removed the exceptions and it looks like its working but not all partitions are listed:

```
PS C:\Windows\system32> Invoke-IcingaCheckUsedPartitionSpace -Verbosity 3
[OK] Used Partition Space: 4 Ok (All must be [OK])
\_ [OK] Partition C: 57.81GiB
\_ [OK] Partition C: 57.81GiB
\_ [OK] Partition D: 611.83GiB
\_ [OK] Partition D: 611.83GiB
| 'partition_d'=656948300000B;;;0;852476900000 'partition_c'=62075750000B;;;0;107375200000
```

Partition E is missing! 

on another server the same, it only shows C and D and all other partition (E,F and G) are not shown by the cmdlet output/performacedata

```
PS C:\Windows\system32> Get-Volume

DriveLetter FileSystemLabel FileSystem DriveType HealthStatus OperationalStatus SizeRemaining      Size
----------- --------------- ---------- --------- ------------ ----------------- -------------      ----
D           Volume          NTFS       Fixed     Healthy      OK                     182.1 GB    200 GB
E           VM              NTFS       Fixed     Healthy      OK                    358.04 GB 593.93 GB
C           System          NTFS       Fixed     Healthy      OK                     42.19 GB    100 GB
```

and BTW: i'm missing the percentage usage for the partitions in the performancedata (in 1.4.1 it showed the percentage and byte values), is there a reason why you removed this or is it a bug?

--- a/doc/31-Changelog.md
+++ b/doc/31-Changelog.md
@@ -13,6 +13,7 @@ Released closed milestones can be found on [GitHub](https://github.com/Icinga/ic
 
 ## Bugfixes
 
+* [#187](https://github.com/Icinga/icinga-powershell-plugins/issues/187) Fixes `Invoke-IcingaCheckUsedPartitionSpace` and `Invoke-IcingaCheckDiskHealth` in case mirrored disks are used within the system, causing an exception on duplicate partition id's
 * [#200](https://github.com/Icinga/icinga-powershell-plugins/issues/200) Fixes `UNKNOWN` for `Invoke-IcingaCheckUsedPartitionSpace`, in case the main partition has no space left which should return `CRITICAL` instead
 * [#233](https://github.com/Icinga/icinga-powershell-plugins/pull/233) Fixes used partition space plugin performance by fetching only partition space data instead of entire disk information collection and renamed it to represent the new method of being able to toggle between free and used space for partitions
 * [#235](https://github.com/Icinga/icinga-powershell-plugins/pull/235) Fixes operational status monitoring output for `Invoke-IcingaCheckDiskHealth`

--- a/provider/disks/Get-IcingaDiskPartitionAssignment.psm1
+++ b/provider/disks/Get-IcingaDiskPartitionAssignment.psm1
@@ -50,6 +50,11 @@ function Get-IcingaDiskPartitionAssignment()
         }
 
         $PartitionInformation[$DiskId].DriveLetters += $PartitionReference;
+
+        if ($PartitionInformation[$DiskId].Partitions.ContainsKey($PartitionId)) {
+            continue;
+        }
+
         $PartitionInformation[$DiskId].Partitions.Add(
             $PartitionId, $PartitionReference
         );

